TriblerGUI/tribler_request_manager.py

```diff
@@ -215,29 +90,26 @@ def update_worker_settings(self, protocol=None, host=None, port=None, key=None):
         self.default_port = port
         self.default_key = key
 
-    def perform_request(self, request_manager, endpoint, reply_callback, data, method):
+    def dispatch_request(self, priority, request_manager, endpoint, reply_callback, data, method):
         self.num_requests += 1
         worker_index = self.num_requests % self.pool_size
 
         num_worker = len(self.request_workers)
         if num_worker < self.pool_size:
             for _ in range(self.pool_size - num_worker):
-                worker = TriblerRequestWorker()
-                self.request_workers.append(worker)
-                if self.default_protocol:
-                    worker.update_protocol(self.default_protocol)
-                if self.default_host:
-                    worker.update_host(self.default_host)
-                if self.default_port:
-                    worker.update_port(self.default_port)
-                if self.default_key:
-                    worker.update_key(self.default_key)
-
-        network_reply = self.request_workers[worker_index].perform_request(endpoint, reply_callback, data, method)
+                self.request_workers.append(
+                    TriblerRequestWorker(
+                        protocol=self.default_protocol,
+                        host=self.default_host,
+                        port=self.default_host,
+                        key=self.default_key,
+                    )
+                )
+        network_reply = self.request_workers[worker_index].do_request(priority, endpoint, reply_callback, data, method)
         request_manager.set_reply_handle(network_reply)
 
     def download_file(self, request_manager, endpoint, reply_callback):
-        self.perform_request(request_manager, endpoint, reply_callback, "", method="GET")
+        self.dispatch_request(request_manager, endpoint, reply_callback, "", method="GET")
```

Comment:
Shouldn't this take a `priority` argument?

Reply:
Probably. I'm going to refactor this further.